webserver/Webserver.go

```diff
@@ -1,16 +1,63 @@
 package webserver
 
 import (
+	"context"
 	"crypto/tls"
+	"errors"
 	"fmt"
-	"github.com/peterfromthehill/tproxy/services"
 	"log"
 	"net"
 	"net/http"
+	"os"
+	"reflect"
+	"syscall"
+	"unsafe"
+
+	"github.com/peterfromthehill/tproxy/services"
+	"go.elastic.co/apm"
 )
 
 type Webserver struct {
-	HttpPort, HttpsPort string
+	HttpPort, HttpsPort int
+	SSLCert, SSLKey     string
+}
+
+type contextKey struct {
+	key string
+}
+
+var connContextKey = &contextKey{"http-conn"}
+
+func saveConnInContext(ctx context.Context, c net.Conn) context.Context {
+	return context.WithValue(ctx, connContextKey, c)
+}
+func getConn(r *http.Request) net.Conn {
+	return r.Context().Value(connContextKey).(net.Conn)
+}
+
+// getConnFromTLSConn returns the internal wrapped connection from the tls.Conn.
+func getConnFromTLSConn(tlsConn *tls.Conn) net.Conn {
+	// XXX: This is really BAD!!! Only way currently to get the underlying
```

Comment:
Can you remove the "XXX: This is really BAD!!! " part of the comment and instead using "FIXME: ..."